From d5032288a2a00e5ffe6ef7f384ce140cff7639c3 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 24 Nov 2022 09:15:40 +0200
Subject: [PATCH] [SW-107110] gaudi3: reset: add last reset cause

This patch adds support to print the last reset
cause for gaudi3. The existing __weak definition
of rst_src_dump is modified so that this function
can be used for all platforms.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I61890a3aa3a35b078c0008536678a4fb58b791c5
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 037dd48..3f62e54 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -128,7 +128,7 @@
 
 }
 
-__weak void rst_src_dump(void)
+void rst_src_dump(void)
 {
 	unsigned int rst_src = hl_rst_src_get();
 	char *cause;
@@ -149,6 +149,18 @@
 	case HL_SW_ALL_RST:
 		cause = "software all";
 		break;
+	case HL_SOFT_RST:
+		cause = "soft";
+		break;
+	case HL_FW_ALL_RST:
+		cause = "firmware all";
+		break;
+	case HL_FLR_RST:
+		cause = "functional";
+		break;
+	case HL_ECC_DERR_RST:
+		cause = "ecc derr";
+		break;
 	default:
 		cause = "unknown";
 		break;
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 30213d7..2ce57cc 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -757,48 +757,6 @@
 			mmDCORE3_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
 }
 
-void rst_src_dump(void)
-{
-	unsigned int rst_src = hl_rst_src_get();
-	const char *cause;
-
-	switch (rst_src) {
-	case HL_COLD_RST:
-		cause = "cold";
-		break;
-	case HL_MANUAL_RST:
-		cause = "manual";
-		break;
-	case HL_PRSTN_RST:
-		cause = "prstn";
-		break;
-	case HL_SOFT_RST:
-		cause = "soft";
-		break;
-	case HL_WD_RST:
-		cause = "watch dog";
-		break;
-	case HL_FW_ALL_RST:
-		cause = "firmware all";
-		break;
-	case HL_SW_ALL_RST:
-		cause = "software all";
-		break;
-	case HL_FLR_RST:
-		cause = "functional";
-		break;
-	case HL_ECC_DERR_RST:
-		cause = "ecc derr";
-		break;
-	default:
-		cause = "unknown";
-		break;
-	}
-	preboot_puts_raw("Last reset cause is: ");
-	preboot_puts_raw(cause);
-	preboot_puts(" reset\n");
-}
-
 void preboot_read_straps(void)
 {
 	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 604df9f..2a537b3 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -154,44 +154,6 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
 }
 
-void rst_src_dump(void)
-{
-	unsigned int rst_src = hl_rst_src_get();
-	char *cause;
-
-	switch (rst_src) {
-	case HL_COLD_RST:
-		cause = "cold";
-		break;
-	case HL_MANUAL_RST:
-		cause = "manual";
-		break;
-	case HL_PRSTN_RST:
-		cause = "prstn";
-		break;
-	case HL_SOFT_RST:
-		cause = "soft";
-		break;
-	case HL_WD_RST:
-		cause = "watch dog";
-		break;
-	case HL_FW_ALL_RST:
-		cause = "firmware all";
-		break;
-	case HL_SW_ALL_RST:
-		cause = "software all";
-		break;
-	case HL_FLR_RST:
-		cause = "functional";
-		break;
-	default:
-		cause = "unknown";
-		break;
-	}
-	preboot_puts_raw("Last reset cause is: ");
-	preboot_puts_raw(cause);
-	preboot_puts(" reset\n");
-}
 void preboot_init_pcie_phy_fw_update(void)
 {
 	/* Configure FW update size */
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index 6095ec0..aa3b8aa 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -58,6 +58,8 @@
 	HL_FW_ALL_RST = 5,
 	HL_SW_ALL_RST = 8,
 	HL_FLR_RST = 9,
+	/* added for compilation */
+	HL_ECC_DERR_RST,
 };
 
 #define SHIFT_4K_POS		12
